packages/groq-builder/src/commands/deref.test.ts

```diff
@@ -17,11 +17,12 @@ describe("deref", () => {
   const qVariants = qVariantsRefs.deref();
 
   it("should deref a single item", () => {
-    expectTypeOf<
-      InferResultType<typeof qCategory>
-    >().toEqualTypeOf<SanitySchema.Category | null>();
+    // I don't understand why this is now failing.
+    // switching to toMatchTypeOf seems to fix it, but that does not seem like the correct solution
+    type CategoryInferenceResult = InferResultType<typeof qCategory>;
+    expectTypeOf<CategoryInferenceResult>().toEqualTypeOf<SanitySchema.Category | null>();
```

Comment:
@scottrippey This failure is really throwing me for a loop. I cannot understand why this is failing. I know you're busy on a client project, but if you get time I could use some guidance on what might be happening here. Otherwise, I can try and take another stab at it next week once I've had a bit of a brain break.

Upgrading to vitest caused a few other problems when using `expectTypeOf` in addition to `toEqualTypeOf` and it is unclear to me if this is due to it finding an actual problem with our types, or if they have a bug on their end (vitest's).

I've googled this quite a bit and not coming back with any promising leads.

[Here's a link to one of the failures](https://github.com/FormidableLabs/groqd/actions/runs/8302614759/job/22725239327#step:9:19)

It appears if I convert the test to check an array of categories like the variants test below, it all works fine. But something to do with the slice operation and checking a single item seems to cause issues.

Reply:
Debugging these type errors is hard. Vitest gives a very cryptic error message ("expected 1 argument but got 0").
This error simply means that the types didn't match.
To find the actual issue, you've got to do some debugging in TypeScript, which is a unique experience. I'll take a look locally and see what I can find!

Reply:
I found the issue, it looks like with this update,  `{ foo: any } & { bar: any }` is no longer the same as `{ foo: any, bar: any }`.
I improved one of my types upstream, and it fixed this problem here.

Reply:
@scottrippey Thanks for taking a look. I think we were pushing at the same time, so didn't notice until now but your commit appears to have fixed that issue, but created a new type error in the build step :(.

```
src/tests/mocks/nextjs-sanity-fe-mocks.ts:130:3 - error TS4053: Return type of public method from exported class has or is using name 'decorator' from external module "/Users/sawyerburnett/git-repos/formidable/groqd/packages/groq-builder/src/tests/schemas/nextjs-sanity-fe.generated" but cannot be named.

130   generateSeedData({
      ~~~~~~~~~~~~~~~~

src/tests/mocks/nextjs-sanity-fe-mocks.ts:130:3 - error TS4053: Return type of public method from exported class has or is using name 'referenced' from external module "/Users/sawyerburnett/git-repos/formidable/groqd/packages/groq-builder/src/tests/schemas/nextjs-sanity-fe.generated" but cannot be named.

130   generateSeedData({
      ~~~~~~~~~~~~~~~~


Found 2 errors in the same file, starting at: src/tests/mocks/nextjs-sanity-fe-mocks.ts:130
```

I did go back to detached state with your commit and verified when running `pn run build` on `/groq-builder` we now get that error. I will see if I can find a solution to it. It seems decorator is a reserved word and is now causing an issue 🤷🏻